src/Donations.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { formatCurrency, getDonations, Donation, track } from './utils';
+
+type State = {
+  loading: boolean;
+  donations: Donation[];
+  totalAmount: number;
+};
+
+class Donations extends Component {
+  state: State = {
+    loading: true,
+    donations: [],
+    totalAmount: 0,
+  };
+
+  async componentDidMount() {
+    const donations = this.state.donations;
+
+    const result = await getDonations();
+
+    for (var i = 0; i < result.length; i++) {
+      donations.push(result[i]);
+    }
+
+    this.setState({
+      donations: donations,
+      loading: false,
+      totalAmount: this.state.donations.reduce((p, c) => p + c.amount, 0),
+    });
+  }
+
+  renderDonation(donation: Donation) {
+    return (
+      <div>
+        <b style={{ textDecoration: 'underline' }}>
```

Comment:
Styles - don't do them inline where you don't need to! There should be a CSS file for the component